Automattic/wp-calypso
# Editor / CPT: Migrate notices to component using type object label

Related: #5987
Closes #3695

This pull request seeks to refactor editor notices, moving the render logic to a separate component `<EditorNotice />`. In doing so, the notices are now aware of the the labels associated with a post type. Specifically, we now use the `view_item` label of the post type when displaying a "View Post" action after publishing a post.

Before|After
---|---
![Before](https://cloud.githubusercontent.com/assets/1779930/16850470/50bec90a-49cd-11e6-9435-65f60f6e1c64.png)|![After](https://cloud.githubusercontent.com/assets/1779930/16850471/50bff58c-49cd-11e6-9ebc-fa562b47c22f.png)

__Implementation notes:__

I plan to include tests for this new component, but had encountered a few issues in the process of doing so. I'll plan to push up a commit including tests before merging these changes.

__Testing instructions:__

Verify that there are no regressions for all existing notices, including success (published, updated) and failures (no content, unknown failure).

1. Navigate to the [post editor](http://calypso.localhost:3000/post) (preferably the [portfolio project editor](http://calypso.localhost:3000/edit/jetpack-portfolio) if enabled to demonstrate view label)
2. Add a title and/or content
3. Publish the post
4. Note that the notice shown appears exactly as it does in master for posts, or using the `view_item` label if applicable for a custom post type

Test live: https://calypso.live/?branch=add/editor-notices
